qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -333,13 +355,130 @@ def _mapper(self, circuit_graph, coupling_graph, trials=20):
+        depth_cnt = [bdag.depth() for bdag in block_dags]
+        maxind = np.argmax(depth_cnt)
+        for i, dag in enumerate(block_dags):
+            if i == maxind:
+                block_circuits.append(dag_to_circuit(dag))
+            else:
+                layout_xform = LayoutTransformation(
+                    new_coupling,
+                    block_layouts[i],
+                    block_layouts[maxind],
+                )
+                match_dag = layout_xform.run(dag)
+                block_circuits.append(dag_to_circuit(match_dag))
+
+        final_permutation = combine_permutations(
+            list(current_layout.get_physical_bits().keys()),
+            list(block_layouts[maxind].get_physical_bits().keys()),
+        )
+        final_layout = Layout.from_intlist(final_permutation, *self.qregs.values())
+        return cf_op.replace_blocks(block_circuits), final_layout
+
+    def _transpile_controlflow_looping(self, cf_op, current_layout):
+        """for looping this pass adds a swap layer at the end of the loop body to bring
+        the layout back to the expected starting layout. This could be reduced a bit by
+        specializing to for_loop and while_loop. If the loop body contains a "continue_loop", the
+        swap layer is placed before the continue". No check is done to make sure "continue" is final
+        statement in loop block.
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.transpiler.passes.routing import LayoutTransformation
+
+        def _move_continue_to_end(dag):
+            """Check if continue exists in block. If it does move it to the end of the circuit
+            so layout_xform doesn't get skipped."""
+            continue_nodes = dag.named_nodes("continue_loop")
+            if continue_nodes:
+                if len(continue_nodes) > 1:
+                    raise CircuitError("Multiple 'continue' statements contained in loop")
+                if len(continue_nodes) <= 1:
+                    c_node = continue_nodes[-1]
+                    dag.remove_op_node(c_node)
+                    dag.apply_operation_back(c_node.op, c_node.qargs, c_node.cargs)
+            return dag
+
+        new_coupling = layout_transform(self.coupling_map, current_layout)
+        dag_block = circuit_to_dag(cf_op.blocks[0])
+        _pass = self.__class__(new_coupling, initial_layout=None, seed=self.seed)
+        start_qreg = QuantumRegister(len(self._qubit_indices), "q")
+        start_layout = Layout.generate_trivial_layout(start_qreg)
+        updated_dag_block = _pass.run(dag_block)
+        updated_layout = _pass.property_set["final_layout"].copy()
+
+        layout_xform = LayoutTransformation(new_coupling, updated_layout, start_layout)
+        match_dag = layout_xform.run(updated_dag_block)
+        match_dag = _move_continue_to_end(match_dag)
+
+        match_circ = dag_to_circuit(match_dag)
+        return cf_op.replace_blocks([match_circ]), current_layout
```

Comment:
As pointed out by Kevin on the BasicSwap pr, without the move the continue statement in the compiled block appears before the swaps from the layout transformation which bring the final layout to where it needs to be. It is assumed there is only one continue in the block although I think the logic handling is not complete as you have pointed out. 

Since we are putting off break and continue handling until later, however, this has been removed.

By the way it was the usage of Aer in tests for this PR which caught the error for this PR (in addition to Kevin in BasicSwap).